### PR TITLE
Renaming `base cluster` to `cluster template`

### PR DESCRIPTION
**Pull request by jayanth-tjvrr**
Fixes: #345 
This PR includes renaming of `base cluster` to `cluster manifest` everywhere in the docs of arlon and also clustertemplate alias has been added to basecluster command. So, now a user can a command as `arlon clustertemplate preparegit ....` and `arlon clustertemplate validategit ....`.

### CODE REVIEW

File: docs/clustertemplate.md [hunk 1]
```diff
@@ -1,17 +1,17 @@
-# Next-gen Cluster Provisioning using Base Clusters
+# Next-gen Cluster Provisioning using cluster template
```
Comment (bcle) [inline docs/clustertemplate.md:1]: Captalize and add 's': **Cluster Templates**

File: docs/clustertemplate.md [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
 
 This proposal describes a new way of provisioning workload clusters in Arlon.
-The most significant change is the *Base Cluster* construct, which replaces the current ClusterSpec.
+The most significant change is the *Cluster Template* construct, which replaces the current ClusterSpec.
 To distinguish them from current generation
-clusters, the ones deployed from a base cluster are called next-gen clusters.
+clusters, the ones deployed from a cluster template are called next-gen clusters.
```
Comment (bcle) [inline docs/clustertemplate.md:6]: add "or gen2 clusters"

**Review by bcle — APPROVED** [review]
Some minor nits to address before merging.

File: docs/gen2_overrides_proposal_3.md [hunk 3]
```diff
@@ -60,21 +60,21 @@ This is an update to proposal of [Gen2 cluster overrides proposal-2](gen2_overri
...
+- This can be an added feature aw well in one way because whenever there is a change in the cluster template configuration, it won't be immediately picked up. This brings user the ability to promote cluster template changes sensibly through each of our environments.
 - The cluster created using overrides approach are not completely declarative.
 
 ### UX (User experience):
 
 A user can pass the patch file for the cluster as an argument while executing the `arlon cluster create ..` command with the --override flag. The command would look like:
 
-`arlon cluster create <cluster-name> --repo-url <repo url where base manifest is present> --repo-path <repo path to the base manifest> --override <path to the patch file> --patch-repo-url <repo url where patch files should be stored> --patch-repo-path <repo path to store the patch files>` 
+`arlon cluster create <cluster-name> --repo-url <repo url where cluster template is present> --repo-path <repo path to the cluster template> --override <path to the patch file> --patch-repo-url <repo url where patch files should be stored> --patch-repo-path <repo path to store the patch files>`
```
Comment (cruizen) [inline docs/gen2_overrides_proposal_3.md:77]:
```suggestion
`arlon cluster create <cluster-name> --repo-url <repo url where cluster template is present> --repo-path <repo path to the cluster template> --override <path to the patch file> --patch-repo-url <repo url where patch file should be stored> --patch-repo-path <repo path to store the patch file>` 
```

File: docs/clustertemplate.md [hunk 4]
```diff
@@ -103,25 +102,25 @@ The `--profile` flag is optional; a cluster can be created with no profile.
...
 
-The application's spec uses a ApplicationSourceKustomize that points to the base cluster's git
+The application's spec uses a ApplicationSourceKustomize that points to the bcluster template's git
 directory. The spec ensures that all deployed resources are configured to:
 
 - Reside in the `cluster-a` namespace, which is deployed by the *arlon app* (see below).
 This achieved by setting `app.Spec.Destination.Namespace` to the workload cluster's name
   (*this only works if the resources do not specify an explicit namespace; this requirement is
-taken care of by the "prep" step on the base cluster*).
+taken care of by the "prep" step on the cluster template*).
```
Comment (cruizen) [inline docs/clustertemplate.md:121]:
```suggestion
taken care of by the "prep" step on the *cluster template*).
```

File: docs/clustertemplate.md [hunk 5]
```diff
@@ -1,17 +1,16 @@
-# Next-gen Cluster Provisioning using Base Clusters
+# Next-gen Cluster Provisioning using Cluster templates
 
 This proposal describes a new way of provisioning workload clusters in Arlon.
```
Comment (cruizen) [inline docs/clustertemplate.md:3]:
```suggestion
This document describes a new way of provisioning workload clusters in Arlon.
```
This is no longer a proposal

File: docs/clustertemplate.md [hunk 6]
```diff
@@ -1,17 +1,16 @@
...
-The most significant change is the *Base Cluster* construct, which replaces the current ClusterSpec.
-To distinguish them from current generation
-clusters, the ones deployed from a base cluster are called next-gen clusters.
+The most significant change is the *Cluster Template* construct, which replaces the current ClusterSpec.
+To distinguish them from current generation, the ones deployed from a cluster template are called next-gen clusters or gen2 clusters.
```
Comment (cruizen) [inline docs/clustertemplate.md:5]:
```suggestion
To distinguish them from the older gen1 clusters, the ones deployed from a cluster template are called next-gen clusters or gen2 clusters.
```

**Review by cruizen — CHANGES_REQUESTED** [review]

File: docs/gen2_overrides_proposal_1.md [hunk 7]
```diff
@@ -27,28 +27,28 @@ So, continuing with the 2nd approach, in the kustomize approach, we create an ov
...
         |- usw2-cluster2
         |- use2-cluster1
 
-The base directory will store the base manifest for the final Cluster API manifests, as well as a kustomization.yaml file that identifies these Cluster API manifests as resources for kustomize to use.
+The clustertemplate directory will store the cluster template for the final Cluster API manifests, as well as a kustomization.yaml file that identifies these Cluster API manifests as resources for kustomize to use.
 
-The contents of kustomize file in base folder is as follows:
+The contents of kustomize file in clustertemplate folder is as follows:
 
 ```
```
Comment (cruizen) [inline docs/gen2_overrides_proposal_1.md:46]:
```suggestion
```yaml
```

File: docs/gen2_overrides_proposal_1.md [hunk 8]
```diff
@@ -1,20 +1,20 @@
 # Gen2 Cluster Overrides - Proposal 1
```
Comment (cruizen) [inline docs/gen2_overrides_proposal_1.md:1]:
```suggestion
# Gen2 Cluster Overrides - Proposal 1

```

**Review by cruizen — APPROVED** [review]
Approved. Take a look at minor markdown lint suggestions.

**Pull request merged**